### PR TITLE
.shellcheckrc, tests/lib: fix unreachable code warnings without global disable

### PR DESCRIPTION
**Pull request by ernestl**
Shellcheck v0.9.0 added [SC2317](https://github.com/koalaman/shellcheck/commit/642ad8612597b28af4026bde289985583fddb69d) to detect unreachable code.
This caused shellcheck failures that is already addressed with [this approach](https://github.com/ernestl/snapd/commit/1d417aa3d0ed4a105162840412ef57d49dd97951).

This proposal is an alternative approach to avoid global disable of the unreachable code check:
1. Excluding the error at specific places as required for placeholder scripts that need to exit with failure if not populated (tests/lib/pkgdb.sh)
2. Avoiding the error by disabling unused code in if statement (tests/lib/prepare.sh)

### CODE REVIEW

**Review by pedronis — COMMENTED** [review]
I'm not against this, the best persons to review this are probably @sergiocazzolato who mainly takes care of these scripts, and @MiguelPires who did the original fixing

File: tests/lib/prepare.sh [hunk 1]
```diff
@@ -735,28 +735,29 @@ EOF
...
-
-        #shellcheck disable=SC2010
-        kver=$(ls "config"-* | grep -Po 'config-\K.*')
-
-        # depmod assumes that /lib/modules/$kver is under basepath
-        mkdir -p fake/lib
-        ln -s "$PWD/modules" fake/lib/modules
-        depmod -b "$PWD/fake" -A -v "$kver"
-        rm -rf fake
+        local uc20_is_ready=false
```
Comment (pedronis) [inline tests/lib/prepare.sh:738]: this probably needs a comment about why the code is structured this way, otherwise it's a bit begging for questions

**Comment by MiguelPires** [comment]
@ernestl can this be marked "Ready for review"?

File: tests/lib/prepare.sh (hunk 1 above)
Comment (ernestl) [reply tests/lib/prepare.sh]: Added explanation comment, please let me know if that is what you had in mind.

File: tests/lib/prepare.sh [hunk 2]
```diff
@@ -735,28 +735,33 @@ EOF
...
-
-        #shellcheck disable=SC2010
-        kver=$(ls "config"-* | grep -Po 'config-\K.*')
 
-        # depmod assumes that /lib/modules/$kver is under basepath
-        mkdir -p fake/lib
-        ln -s "$PWD/modules" fake/lib/modules
-        depmod -b "$PWD/fake" -A -v "$kver"
-        rm -rf fake
+        # To avoid shellcheck unused code warning, we cannot use "exit 0" to disable
```
Comment (mvo5) [inline tests/lib/prepare.sh:739]: Thanks for the comment - this is fine now and I expect a followup shortly that drops the conditional to actually enable this code, it seems to be ready and useful.

**Review by mvo5 — APPROVED** [review]
Thank you

**Review by MiguelPires — APPROVED** [review]
LGTM

**Review by sergiocazzolato — APPROVED** [review]
Nice, thanks for this

**Pull request merged**